polaris/hub/client.py

```diff
@@ -662,18 +720,193 @@ def upload_benchmark(
+    ) -> Dict[str, Response]:
+        """Upload a competition to the Polaris Hub.
+
+        Args:
+            competition: The competition to upload.
+            timeout: Request timeout values. User can modify the value when uploading large dataset as needed.
+                This can be a single value with the timeout in seconds for all IO operations, or a more granular
+                tuple with (connect_timeout, write_timeout). The type of the the timout parameter comes from `httpx`.
+                Since datasets can get large, it might be needed to increase the write timeout for larger datasets.
+                See also: https://www.python-httpx.org/advanced/#timeout-configuration
+            owner: Which Hub user or organization owns the artifact. Takes precedence over `dataset.owner`.
+            if_exists: Action for handling existing files in the Zarr archive. Options are 'raise' to throw
+                an error, 'replace' to overwrite, or 'skip' to proceed without altering the existing files.
+
+        Returns:
+            A object containing responses from the Polaris Hub regarding both the dataset and benchmark upload.
+        """
+        ACCESS = "private"
+
+        if competition.dataset.zarr_root is not None:
+            logger.info("Zarr-based competitions are not supported at this time")
+            return
+
+        # Upload competition dataset
+        dataset_response = self._upload_dataset(
+            competition.dataset, ArtifactType.COMPETITION.value, ACCESS, timeout, owner, if_exists
+        )
+        # Upload competition benchmark
+        competition_response = self._upload_benchmark(
+            competition, ArtifactType.COMPETITION.value, ACCESS, owner
+        )
+
+        return {
+            "dataset_response": dataset_response,
+            "competition_response": competition_response,
+        }
+
+    def get_competition(
+        self, owner: Union[str, HubOwner], name: str, verify_checksum: bool = True
+    ) -> CompetitionSpecification:
+        """Load a competition from the Polaris Hub.
+
+        Args:
+            owner: The owner of the competition. Can be either a user or organization from the Polaris Hub.
+            name: The name of the competition.
+            verify_checksum: Whether to use the checksum to verify the integrity of the dataset.
+
+        Returns:
+            A `CompetitionSpecification` instance, if it exists.
+        """
+
+        response = self._base_request_to_hub(url=f"/v2/competition/{owner}/{name}", method="GET")
+
+        # TODO (cwognum): Currently, the competition endpoints do not return the owner info for the underlying dataset.
+        # TODO (jstlaurent): Use the same owner for now, until the competition returns a better dataset entity
+        response["dataset"] = self._get_dataset(
+            owner, response["dataset"]["name"], ArtifactType.COMPETITION, verify_checksum=verify_checksum
+        )
```

Comment:
**Suggestion**
The Hub now does return the proper owner for the related dataset:
```suggestion
        # TODO (jstlaurent): response["dataset"]["artifactId"] is the owner/name unique identifier,
        #  but we'd need to change the signature of get_dataset to use it
        response["dataset"] = self._get_dataset(
            response["dataset"]["owner"]["slug"], 
            response["dataset"]["name"], 
            ArtifactType.COMPETITION, 
            verify_checksum=verify_checksum
        )
```